lxd/network/driver_ovn.go

```diff
@@ -2098,9 +2098,22 @@ func (n *ovn) InstanceDevicePortAdd(instanceID int, instanceName string, deviceN
 	}
 
 	// Add DNS records for port's IPs, and retrieve the IP addresses used.
-	dnsUUID, dnsIPv4, dnsIPv6, err := client.LogicalSwitchPortSetDNS(n.getIntSwitchName(), instancePortName, fmt.Sprintf("%s.%s", instanceName, n.getDomainName()))
-	if err != nil {
-		return "", err
+	dnsName := fmt.Sprintf("%s.%s", instanceName, n.getDomainName())
+	var dnsUUID string
+	var dnsIPv4, dnsIPv6 net.IP
+
+	// Retry a few times in case port has not yet allocated dynamic IPs.
+	for i := 0; i < 5; i++ {
```

Comment:
So if this runs all 5 times and fails every time, what will return the error to the user?

I would have expected a check of `err != nil` after the loop.